internal/services/securitycenter/security_center_subscription_pricing_resource.go

```diff
@@ -105,12 +147,20 @@ func resourceSecurityCenterSubscriptionPricingUpdate(d *pluginsdk.ResourceData,
 		}
 
 		if existing.Model != nil && existing.Model.Properties != nil && existing.Model.Properties.PricingTier != pricings_v2023_01_01.PricingTierFree {
-			return fmt.Errorf("the princing tier of this subscription is not Free \r %+v", tf.ImportAsExistsError("azurerm_security_center_subscription_pricing", id.ID()))
+			return fmt.Errorf("the pricing tier of this subscription is not Free \r %+v", tf.ImportAsExistsError("azurerm_security_center_subscription_pricing", id.ID()))
 		}
 	}
 
-	if v, ok := d.GetOk("subplan"); ok {
-		pricing.Properties.SubPlan = utils.String(v.(string))
+	if vSub, okSub := d.GetOk("subplan"); okSub {
+		pricing.Properties.SubPlan = utils.String(vSub.(string))
+	}
+
+	// can not set extensions for free tier
+	if pricing.Properties.PricingTier == pricings_v2023_01_01.PricingTierStandard {
+		if vExt, okExt := d.GetOk("extension"); okExt {
+			var extensions = expandSecurityCenterSubscriptionPricingExtensions(vExt.([]interface{}))
+			pricing.Properties.Extensions = extensions
+		}
 	}
```

Comment:
can we use d.HasChanges("extension") here? This will allow users who want to use the default values to use ignore_changes in their config.

Reply:
I preferer to keep it explicit as least on this stage, if this raise in the future again we may add it

Reply:
Ok, Sure, applied